packages/rocketchat-livechat/server/lib/Livechat.js

```diff
@@ -363,13 +365,17 @@ RocketChat.Livechat = {
 		} else if (RocketChat.settings.get('Livechat_Routing_Method') !== 'Guest_Pool') {
 			agent = RocketChat.Livechat.getNextAgent(transferData.departmentId);
 		} else {
+			RocketChat.models.Rooms.changeDepartmentIdByRoomId(room._id, transferData.departmentId);
 			return RocketChat.Livechat.returnRoomAsInquiry(room._id, transferData.departmentId);
 		}
 
 		const servedBy = room.servedBy;
 
 		if (agent && agent.agentId !== servedBy._id) {
```

Comment:
An agent can be in more than one department, so the `agent.agentId` and `servedBy._id` can be the same and is this case the `departmentId` needs to be updated.
The right way to test if the department needs to be updated is checking the `room.departmentId`. If the room  does not have a `departmentId` or `room.departmentId !== transferData.departmentId`, then the department needs to be updated.